### ISSUE TITLE
* * image noop rule not being honoured

### ISSUE THREAD

**Issue by ThurahT** [gorhill/uBlock#522]
Hello, and thanks for a remarkable blocker software. It has indeed replaced both ABE and RPC on my system. Although, I have some issues I don't understand is by design or a malfunction.
I have the following in my ruleset:

"* * 3p block"
"* * image noop"

But no images will actually load. Only from the 1st part page. So, I need to make additional noop rules for all those other pages. like:

"* amazonaws.com * noop"

and so on. Also, I noticed that the rule

"* amazonaws.com image noop"

is ignored all the same.

So having a "* * 3p block" rule becomes a nuisance instead. Many CDN sites only have the images the1st part site needs but allowing them to send *any* content could compromise privacy if there isn't a static filter to pick that out of course.

I use SeaMonkey 2.33.1 and uBlock₀ 1.0.0.0 along with NoScript 2.6.9.31.

TIA.

**Comment by chway** [comment]
Take a look at https://github.com/gorhill/uBlock/wiki/Dynamic-filtering:-rule-syntax, "* * image noop"
 and "* amazonaws.com image noop" are invalid, you need to specify a source-hostname to allow/block images. You should try uMatrix if you want to allow images globally from a specific domain.

Another way to do this:
* * 3p-frame block
* * 3p-script block

Images and CSS will be allowed while scripts and frames will still be blocked.

**Comment by ThurahT** [comment]
Indeed, that seems to be the case... So this is by design?? I thought that a more advanced rule had priority over a less advanced rule as stated in the FAQ/MAN. 

Unfortunately there are more types of data that needs to be blocked than just frames and scripts. That won't cut it.. In that case more types should be added. Like all the types that are visible in the log for instance.

I will take a look on this uMatrix thing even though I want the opposite to what you described... I want to noop all images form *any* domain. Not one specific one..

Thanks for commenting.

**Comment by gorhill** [comment]
> But no images will actually load. Only from the 1st part page. So, I need to make additional noop rules for all those other pages

Currently `image`-based rules are considered less specific than the `3p` rule, hence they have lower priority. I added the `image` row initially strictly for those who wished to save bandwidth. The `3rd-party`-based rows were added for security/privacy purpose. An ambiguity arises when explicit rules exist for both `image` and `3p`, and in such case, I chose to make `3p`-based rules trumps `image`, because security/privacy is far more important than bandwidth-reduction.

@chway is right, uMatrix matches the behavior you are seeking, it's designed to deal with those ambiguities gracefully -- i.e. what you see is what you get.

> I want to noop all images form any domain

uMatrix allows you this (assign a rule to the `image` cell at the top), except that there is no concept of _noop_, because there is no static filtering, it's block or allow.
